src/test/java/ca/corefacility/bioinformatics/irida/utils/NullReplacementDatasetLoader.java

```diff
@@ -0,0 +1,30 @@
+package ca.corefacility.bioinformatics.irida.utils;
+
+import com.github.springtestdbunit.dataset.AbstractDataSetLoader;
+import org.dbunit.dataset.IDataSet;
+import org.dbunit.dataset.ReplacementDataSet;
+import org.dbunit.dataset.xml.FlatXmlDataSet;
+import org.dbunit.dataset.xml.FlatXmlDataSetBuilder;
+import org.springframework.core.io.Resource;
+
+import java.io.InputStream;
+
+/**
+ * Spring DBUnit loader which replaces '[null]' with an actual NULL value in the database.
+ */
+public class NullReplacementDatasetLoader extends AbstractDataSetLoader {
```

Comment:
Added this file because it's otherwise nearly impossible to insert a `NULL` with dbunit. This uses a ReplacementDataSet to replace `[null]` in a dbunit xml file with an actual `NULL` in the object & database.

Reply:
This sounds good. But doesn't dbunit default to `NULL` unless you specify a value? So you could just leave these particular columns empty in the XML (http://dbunit.sourceforge.net/faq.html#flatxmlnull).

Reply:
It does, except that it uses the first entry for a given table it finds in the xml as the schema for what it'll insert (http://dbunit.sourceforge.net/faq.html#differentcolumnnumber).  So if you define:
* `<table1 prop1="1"/>`
* `<table1 prop1=2" prop2="3"/>`
`prop2` will get ignored.  It's a weird dbunit thing that we've had to work around in the past.

In the particular XML I was updating (https://github.com/phac-nml/irida/pull/426/files#diff-194498486cafb928deda1a435ccefa46), it had a few entries of `analysis_submission` already and those ones are tied to **a lot** of tests.  Since `AnalysisSubmission` has a few related subclasses using different properties, it makes it awkward to get the schema for dbunit fully defined for the different types of subclasses.  This `ReplacementDataSet` option makes it easy because now you can fully flesh out the schema for the first item, and insert `NULL` as needed.

Does that make sense?